### ISSUE TITLE
CapData.load_data() duplicate timestamps

### ISSUE THREAD

**Issue by bt-** [pvcaptest/pvcaptest#6]
Consider updating method to remove duplicate timestamps when importing separate files with overlapping dates.

**Comment by UGuntupalli** [comment]
Is this still open ? If yes, I can take a stab at this.

**Comment by bt-** [comment]
@UGuntupalli, yes, this one is open. It would be good to also add sorting by index as well.

**Comment by UGuntupalli** [comment]
@bt-, 
      Thank you. So, a couple of quick questions, just to help me get started: 

```python 
ix_ser = all_sensors.index.to_series()
all_sensors['index'] = ix_ser.apply(lambda x: x.strftime('%m/%d/%Y %H %M'))  # noqa: E501
```
1. Based on the above code, is it fair for me to say that we are assuming that each file will have an index of dates/datetimes ? 
2. Is the intent to drop duplicates after merging or before ? If we want to drop them after, which one would you prefer to retain ?

**Comment by bt-** [comment]
@UGuntupalli, 

>     1. Based on the above code, is it fair for me to say that we are assuming that each file will have an index of dates/datetimes ?
Yes, that is a fair assumption. 

A side note, I would actually like to eventually remove those two lines. They are only there to create an index of dates as strings, which are used in some of the plotting methods, specifically scatter_hv (I think). I made a new issue #59 for that.

>     2. Is the intent to drop duplicates after merging or before ? If we want to drop them after, which one would you prefer to retain ?
I'm assuming you are asking about before or after merging individual csv files, is that correct? What I do now to correct for duplicates is to drop them after merging. That addresses the problem I've run into when you have daily data files, but there is some overlap. It would also drop any duplicates that occur within the daily data file. Dropping before merging would not address if the two files have overlapping indices.  Here is the code I use now:

```
meas.data.sort_index(inplace=True)
meas.data.drop_duplicates(inplace=True)

#reindex so no missing timestamps for clear sky filter
meas.data, missing_intervals, freq_str = util.reindex_datetime(meas.data, report=True)
```
The `reindex_datetime` function is something I've been using, but haven't had a chance to add yet. The sorting and reindexing are definitely things that should be added. I think sorting could be part of the pull request to address this issue, but the reindex should maybe be a separate issue.

**Comment by UGuntupalli** [comment]
@bt- ,
       I finally managed some time to look at this, but I am unable to run the code where ```load_data``` reads the example measured data, however it throws up an error. Here is a screenshot: 
       
![image](https://user-images.githubusercontent.com/47903157/124692156-e3e51f00-de91-11eb-9cea-76bb163bc406.png)

![image](https://user-images.githubusercontent.com/47903157/124692201-fa8b7600-de91-11eb-8c4e-96cf2463f66f.png)


Would you know if the file path or the name needs to be edited for me to try this out ?

**Comment by bt-** [comment]
@UGuntupalli, hmm. Well, the error is indicating the file is not at `../examples/data/example_meas_data.csv`.

I assume that you are running your script from the directory containing the examples directory and that the csv is actually at that path?

It might be easier to review if you create a new branch on your fork to work in and then you can push changes to that branch. A new unit test for this functionality should go into the the [TestLoadDataMethods](https://github.com/UGuntupalli/pvcaptest/blob/30b1a3c9a32b9c91f9cd3b0959877cbd21698541/tests/test_CapData.py#L371) class.

Then, you can create a new folder to hold the csv files with duplicate timestamps in /pvcaptest/tests/data/new_folder.

To merge any changes we'll need a unit test, so this approach should save effort and I'll be able to pull your branch to review.

**Comment by UGuntupalli** [comment]
@bt- , 
          I did create my own branch, but I made zero changes so far. I am current with the master. I am running ```capdata.py``` directly in my IDE. I am using Pycharm,

**Comment by bt-** [comment]
@UGuntupalli, 
Ok, it is hard for me to figure out why the code in your screenshot is not working because I don't have the full context of your local environment. Are you using doctest to run the code in your screenshot? Are you triggering the code to run from within Pycharm? Where is your script you are executing saved relative to the example csv file?

I work with a text editor plus shell, so I'm not familiar with Pycharm, but I'm sure we can get it running.

If you want to establish that your local development branch is functioning as a starting point, one way to do that would be to run the unit tests locally. You can try that by running `pytest tests/` from the project root directory from a command line. If any of those fail, then we'll need to sort that out before doing anything else.

Please try that and let me know if it works or you have any questions. Thanks!

**Comment by bt-** [comment]
I'm closing this due to the refactoring of the data loading in v0.11.1. Data loading was moved to the io module and refactored extensively.  The new [load_data](https://github.com/pvcaptest/pvcaptest/blob/741dd62277b2a4087556a12a1dd4294d0e894753/captest/io.py#L372) function drops duplicate rows by default using the pandas `DataFrame.drop_duplicates`, but this does NOT consider duplicates in the datetime index. 

I looked at the options for dropping duplicate datetime indices briefly after this issue was created and didn't see a straightforward method for determining which of the duplicates to keep.
